packages/flutter_tools/lib/src/android/android_device.dart

```diff
@@ -876,7 +876,13 @@ class _AndroidDevicePortForwarder extends DevicePortForwarder {
         process.throwException('adb did not report forwarded port');
       hostPort = int.tryParse(process.stdout) ?? (throw 'adb returned invalid port number:\n${process.stdout}');
     } else {
-      if (process.stdout.isNotEmpty)
+      // dantup: stdout may be empty or the port we asked it to forward,
+      // depending on some factors I do not understand. On MacOS it's always
+      // empty for Flutter, yet at the terminal it prints the port number
+      // only if it forwarded (nothing if it was already being forwarded).
+      // On CrOS it prints the port number even for Flutter, which results in a
+      // failure unless we allow the hostPort in the output here.
+      if (process.stdout.isNotEmpty && process.stdout.trim() != '$hostPort')
         process.throwException('adb returned error:\n${process.stdout}');
```

Comment:
I'm curious why this `else` branch exists - at this point we've already gotten a valid hostPort. We just chose to fail if we then receive any stdout from `adb`?

We could instead just ignore stdout output.

Reply:
I don't know for sure but I guess we're assuming if we get some unexpected output, it's could be a failure message (some tools write to `stdout`) and it's better to fail and review what's happened than assume that the port was mapping ok (which could result in harder to track down connection failures later in the process).

Reply:
I don't see any mention of the stdout behavior in the adb docs. If this check is only based on observations, than updating it for new observations seem perfectly reasonable.

Reply:
Yeah, I couldn't find any docs describing this (even casually in examples), so my guess is that it was based on how it seemed to work.

I still can't get my head around why Flutter doesn't get the stdout on macOS though, specifically in this case. I did try reading the ADB source code, but it's written in a language I don't speak.

Reply:
The expected failure here is that you've specified an explicit port that's already used, so it can't be opened by `adb`.